infra/gcp/ensure-main-project.sh

```diff
@@ -0,0 +1,173 @@
+enable_api "${PROJECT}" oslogin.googleapis.com
+
+# Make an account for GKE nodes to run as
+color 6 "Creating service account for ${NODES_SVCACCT}"
+ensure_service_account "${PROJECT}" "${NODES_SVCACCT}" "Least-privilege SA for k8s nodes"
+
+color 6 "Empowering ${NODES_SVCACCT} with min permissions"
+acct=$(svc_acct_email "${PROJECT}" "${NODES_SVCACCT}")
+gcloud projects add-iam-policy-binding "${PROJECT}" \
+    --member "serviceAccount:${acct}" \
+    --role roles/logging.logWriter
+gcloud projects add-iam-policy-binding "${PROJECT}" \
+    --member "serviceAccount:${acct}" \
+    --role roles/monitoring.viewer
+gcloud projects add-iam-policy-binding "${PROJECT}" \
+    --member "serviceAccount:${acct}" \
+    --role roles/monitoring.metricWriter
+
+color 6 "Empowering BigQuery admins"
+gcloud projects add-iam-policy-binding "${PROJECT}" \
+    --member "group:${BQ_ADMINS_GROUP}" \
+    --role roles/bigquery.admin
+
+color 6 "Empowering cluster admins"
+gcloud projects add-iam-policy-binding "${PROJECT}" \
+    --member "group:${CLUSTER_ADMINS_GROUP}" \
+    --role roles/compute.viewer
+gcloud projects add-iam-policy-binding "${PROJECT}" \
+    --member "group:${CLUSTER_ADMINS_GROUP}" \
+    --role roles/container.admin
+if ! gcloud --project "${PROJECT}" iam roles describe ServiceAccountLister >/dev/null 2>&1; then
+    gcloud --project "${PROJECT}" --quiet \
+        iam roles create ServiceAccountLister \
+        --title "Service Account Lister" \
+        --description "Can list ServiceAccounts." \
+        --stage GA \
+        --permissions iam.serviceAccounts.list
+fi
+gcloud projects add-iam-policy-binding "${PROJECT}" \
+    --member "group:${CLUSTER_ADMINS_GROUP}" \
+    --role "projects/${PROJECT}/roles/ServiceAccountLister"
+
+color 6 "Empowering GCP accounting"
+gcloud projects add-iam-policy-binding "${PROJECT}" \
+    --member "group:${ACCOUNTING_GROUP}" \
+    --role roles/bigquery.jobUser
+
+color 6 "Creating the BigQuery dataset for billing data"
+if ! bq --project "${PROJECT}" ls "${BQ_BILLING_DATASET}" >/dev/null 2>&1; then
+    bq --project "${PROJECT}" mk "${BQ_BILLING_DATASET}"
+fi
+
+color 6 "Setting BigQuery permissions"
+
+# Merge existing permissions with the ones we need to exist.  We merge
+# permissions because:
+#   * The full list is large and has stuff that is inherited listed in it
+#   * All of our other IAM binding logic calls are additive
+
+CUR=$(tempfile -p k8s-infra-bq-access-cur)
+bq show --format=prettyjson "${PROJECT}":"${BQ_BILLING_DATASET}"  > "${CUR}"
+
+ENSURE=$(tempfile -p k8s-infra-bq-access-new)
+cat > "${ENSURE}" << __EOF__
+{
+  "access": [
+    {
+      "groupByEmail": "${ACCOUNTING_GROUP}",
+      "role": "READER"
+    },
+    {
+      "groupByEmail": "${ACCOUNTING_GROUP}",
+      "role": "roles/bigquery.metadataViewer"
+    },
+    {
+      "groupByEmail": "${ACCOUNTING_GROUP}",
+      "role": "roles/bigquery.user"
+    }
+  ]
+}
+__EOF__
+
+FINAL=$(tempfile -p k8s-infra-bq-access-new)
```

Comment:
I'm not sure `tempfile` is a common program on many Linux distributions and i can't find it for Mac OS. Maybe a warning in a README as prerequisite ?